standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -1441,6 +1480,7 @@ public void commitTxn(CommitTxnRequest rqst) throws NoSuchTxnException, TxnAbort
             // corresponding open txn event.
             LOG.info("Target txn id is missing for source txn id : {} and repl policy {}", sourceTxnId,
                 rqst.getReplPolicy());
+            jdbcTemplate.rollback();
```

Comment:
do you need to call rollback explicitly even when nothing changed

Reply:
Yes, jdbctemplate Txn requires explicit commit or rollback, because this is also a signal that the underlying connection can be released and put back to the pool. When using `@Transactional` annotation or the `TransactionTemplate` functional interface it is done by the framework, but I was using the lower lever direct approach to have fine grained control over txn and conneciton handling.